src/Core/functions.php

```diff
@@ -308,15 +310,35 @@ function deprecationWarning(string $message, int $stackFrame = 1): void
                 $relative
             );
         }
+        $findDeprecatedMethod = function () use ($trace): bool {
+            foreach ($trace as $value) {
+                if (
+                    !isset($value['class']) ||
+                    !isset($value['function']) ||
+                    $value['class'] !== CakeTestCase::class ||
+                    $value['function'] !== 'deprecated'
+                ) {
+                    continue;
+                }
+
+                return true;
+            }
+
+            return false;
+        };
```

Comment:
This will work fine, but should we be doing something 'dumber' and use a `Configure` value, or global variable? `TestCase::deprecated()` could set the global state and `deprecationWarning()` could check if it's set. Using a configure variable of giving users an easy way to have all their deprecations display if they wanted that.

Reply:
@markstory I just added global config, and removed the complexity

>Using a configure variable of giving users an easy way to have all their deprecations display if they wanted that.

yes I added `Configure` , user could enable it to show all duplicated deprecations

Reply:
I did not add `TestCase::showDuplicatedDeprecation` because all people not use cake's testSuite
or if someone want to use cake's repo(s) standalone , or even if they want to show all in web (debug time ) 
having global setting is more useful